### ISSUE TITLE
[4.0.0-beta.0] Huge performance regression in dependency (?)

### ISSUE THREAD

**Issue by csvan** [webpack/webpack-dev-server#2936]
- Operating System: OSX Catalina
- Node Version: 15.5
- NPM Version: 7.2
- webpack Version: 5.11
- webpack-dev-server Version: 4.0.0-beta.0
- Browser: Chrome 87

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

Link to repro: https://github.com/csvan/wds-performance-regression-repro

Sometime over the last 2-3 days, something happened which caused 4.0.0-beta.0 to suddenly increase build times almost exponentially. On our proprietary projects, we have seen startup time go from seconds to over a minute when running the dev server, while running the _exact_ same webpack config outside the dev server is just as fast as it was before.

I am pretty sure this is a problem with a dependency, since 4.0.0-beta.0 has not changed, I simply cannot figure out which one though.

### Expected Behavior

Dev server should be fast

### Actual Behavior

Dev server is slow

### For Bugs; How can we reproduce the behavior?

See https://github.com/csvan/wds-performance-regression-repro

### For Features; What is the motivation and/or use-case for the feature?

To make local development fast

**Comment by csvan** [comment]
A note about the repro - this is a pretty trivial project, and it may _feel_ pretty fast during dev server start (5 seconds on my system). However, that is still 5x slower than running a regular webpack build using the exact same config, and this difference becomes a lot clearer on very large projects (e.g. from abou 10s to almost 2 minutes in one case)

**Comment by csvan** [comment]
Also, FWIW, profiling the run I see it spending most time on e.g.

```
13% building 0/2 entries 1728/2576 dependencies 353/800 modules
```

This is extremely slow in the dev server, whereas when running webpack itself this phase takes only a few seconds. Is there a hook or listener of some sort which the dev server adds which might slow it down?

**Comment by alexander-akait** [comment]
Try to set `static` on any other value, in beta.0 it is `process.cwd()`, so it is very slow (node_modules and etc directories)

**Comment by csvan** [comment]
@alexander-akait thanks, confirming that adding the below brought performance back to what we saw before:

```
  static: [
    {
      directory: path.join(process.cwd(), 'node_modules'),
      watch: false,
    },
  ],
```

I still don't understand what changed/broke though - a regression in Chokidar?? As stated, projects which use 4.0.0-beta.0 but haven't updated their dependencies in about a week are still running fine.

**Comment by alexander-akait** [comment]
By default `watchContentBase` was `false`, not it is true, but we use `process.cwd()` as default value, it is fixed in master, now we use `static` directory (but I want to change it on `public`, because popular), I want to found on new version of webpack-dev-server in the week, (a lot of issues and I am only one who works on many projects)

**Comment by csvan** [comment]
@alexander-akait I see, thanks for all your hard work, it's really appreciated :)

**Comment by frehner** [comment]
> By default `watchContentBase` was `false`, not it is true, but we use `process.cwd()` as default value, it is fixed in master, now we use `static` directory (but I want to change it on `public`, because popular), I want to found on new version of webpack-dev-server in the week, (a lot of issues and I am only one who works on many projects)

an interesting side effect of the defaults currently on `beta-0` are that WDS is also watching for changes to the `.git` directory. I discovered this because VSCode checks for git updates occasionally when you focus it - and that, in turn, updates stuff in git which then causes WDS to hot refresh the page 🙂  

In any case, glad to hear that it's been fixed on master, and I look forward to `beta-1`'s release 👍

**Comment by alexander-akait** [comment]
@frehner yep, fixed in master, just use `static: './public' or something else, otherwise we use `process.cwd()` by default (in `beta-0`) and we look at node_modules/.git/etc directives inside `process.cwd()`
